egress/src/filter.rs

```diff
@@ -0,0 +1,1417 @@
+                                },
+                            );
+                            if let Some((t_list, t_ex)) = times_ex {
+                                time_pri_typ_ts.push((t_ex, ex.priority, type_id, ts_id));
+                                for t_l in t_list {
+                                    time_pri_typ_ts.push((t_l, def_0.priority, type_id, ts_id));
+                                }
+                            } else {
+                                // no overlap just add
+                                time_pri_typ_ts.push((t, def_0.priority, type_id, ts_id));
+                            }
+                        } else {
+                            // no exceptions
+                            time_pri_typ_ts.push((t, def_0.priority, type_id, ts_id));
+                        }
+                    }
+                }
+            }
+
+            if time_pri_typ_ts.is_empty() {
+                // should this happen?
+                warn!("length of 0 for this label {:?}", label);
+            } else {
+                // sort the list by priority
+                time_pri_typ_ts.sort_by_key(|item| (item.1));
+                //eprintln!("sorted temp prioritites: {time_pri_typ_ts:?}");
+                // get first and last
+                let first_time = time_pri_typ_ts
+                    .iter()
+                    .min_by_key(|item| (item.0.from))
+                    .unwrap()
+                    .0
+                    .from;
+                let last_time = time_pri_typ_ts
+                    .iter()
+                    .min_by_key(|item| (item.0.to))
+                    .unwrap()
+                    .0
+                    .to;
+                // keep looping until no more holes...
+                filter.insert(
+                    label,
+                    fill_holes(
+                        time_pri_typ_ts,
+                        Timerange {
+                            from: first_time,
+                            to: last_time,
+                        },
+                    ),
+                );
+            }
+        }
+    }
+    // sort by descending from time since otherwise unordered
+    // removing this will cause tests to fail... if its ok for other stuff could potentially be moved into the test framework somehow?
+    for list in filter.values_mut() {
+        list.sort_by_key(|item| (item.from));
+    }
+    eprintln!("length filter: {:?}", filter.len());
+    Ok(filter)
+}
+
+pub async fn get_filter(
+    conn: &PooledPgConn<'_>,
+    from: DateTime<Utc>,
+    to: DateTime<Utc>,
+    filter_table: FilterTimeseriesTableLock,
+    label: FilterLabel,
+) -> Result<Option<Vec<FilterData>>, tokio_postgres::Error> {
+    // get the background filter list, and lookup this label
+    // have to clone, otherwise does not work...
+    let ft = filter_table
+        .read()
+        .map_err(|e| Error::Lock(e.to_string()))
+        .unwrap()
+        .clone();
```

Comment:
Don't think we need the clone and the unwrap
```suggestion
) -> Result<Option<Vec<FilterData>>, Error> {
    // get the background filter list, and lookup this label
    // have to clone, otherwise does not work...
    let ft = filter_table
        .read()
        .map_err(|e| Error::Lock(e.to_string()))?;
```